target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -243,3 +350,106 @@ impl crate::accounts::PostUpdates {
         }
     }
 }
+
+struct VaaComponents {
+    verified_signatures: u8,
+    emitter_address:     [u8; 32],
+    emitter_chain:       u16,
+}
+
+fn post_price_update_from_vaa<'info>(
```

Comment:
This is a copy paste refactor to avoid duplicating this code in :
`post_updates` and `post_updates_atomic`